### PR TITLE
Simplify and unify examples

### PR DESCRIPTION
**Pull request by BenjaminW3**
* Replaces most of the uses of `auto` with the correct type (see #490)
* Unifies the naming of variables, types and comments
* simplifies some examples

### CODE REVIEW

File: example/bufferCopy/src/bufferCopy.cpp [hunk 1]
```diff
@@ -28,8 +28,7 @@
 
 #define DIMENSION 3
 
-//#############################################################################
-
+    //-----------------------------------------------------------------------------
```
Comment (ax3l) [inline example/bufferCopy/src/bufferCopy.cpp]: what does this line annotate? Is the indent intentional?

File: example/bufferCopy/src/bufferCopy.cpp [hunk 2]
```diff
@@ -168,52 +138,35 @@ struct FillBufferKernel
...
 -> int
 {
-    // Configure types
+    // Define the index domain
     using Dim = alpaka::dim::DimInt<DIMENSION>;
     using Idx = std::size_t;
-    using Extents = Idx;
-    using Host = alpaka::acc::AccCpuSerial<Dim, Idx>;
+
+    // Define the accelerator types
```
Comment (ax3l) [inline example/bufferCopy/src/bufferCopy.cpp]: maybe without ~~types~~

File: example/bufferCopy/src/bufferCopy.cpp [hunk 3]
```diff
@@ -237,128 +191,108 @@ auto main()
...
-#endif
-#if DIMENSION > 1
-        static_cast<Idx>(nElementsPerDim),
-#endif
-        static_cast<Idx>(nElementsPerDim));
+    const Vec extents(Vec::all(static_cast<Idx>(nElementsPerDim)));
 
+    // Allocate host memory buffers
+    //
+    // The `alloc` method returns a reference conted buffer handle.
```
Comment (ax3l) [inline example/bufferCopy/src/bufferCopy.cpp]: typo: counted

File: example/bufferCopy/src/bufferCopy.cpp [hunk 4]
```diff
@@ -237,128 +191,108 @@ auto main()
...
-    // directly, but it can return the
-    // pointer to the memory within the
-    // buffer (getPtrNative). This pointer
-    // can be used to write some values into
-    // the buffer memory. Mind, that only a host
-    // can write on host memory. The same holds
-    // for device memory.
-    for(size_t i(0); i < extents.prod(); ++i)
+    // You can not access the inner
+    // elements of a buffer diretly, but
```
Comment (ax3l) [inline example/bufferCopy/src/bufferCopy.cpp]: typo: directly

**Comment by psychocoderHPC** [comment]
@BenjaminW3  Please wait always for a review.

**Pull request merged**